### PR TITLE
Feature/user auth report

### LINKED ISSUE #261: User Authentication Report
## Feature request
User authentication report

### Description
Include a user Authentication report in the Report section.

### Implementation considerations
The report should cover the logins, logout, login failures, etc.

The field of this report should include: 
* username
* role
* datetime
* device (?)
* OS (?)
* IP Addr (?)
* Mac Addr(?)

More fields could be added in case of need.

A table should be created in the UI to display the information. 

An Endpoint and its corresponding filter logic should be implemented in the backend.

Verify how to access keycloack config https://keycloak.discourse.group/t/how-to-make-login-events-into-a-log-file/264/2

### PR DESCRIPTION
**Pull request by matiasbavera**
## What's new
fixes #261

* Created a dockerfile for Keycloak and added a json logger pluggin
* Created a fluentd configmap to send keycloak information to the reporting server
* Created Login, Logout and Login Failures reports. Endpoints and UI.
* Updated deploy instructions
* Updated bootstrap keycloak

## Self-checks

- [X] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [X] I added unit-tests for new components
- [X] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

File: example-deployment/docker/keycloak/keycloak.dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM quay.io/keycloak/keycloak:12.0.4
+
+COPY ./jsonlog-event-listener.jar /opt/jboss/keycloak/standalone/deployments/jsonlog-event-listener.jar
```
Comment (koonpeng) [inline example-deployment/docker/keycloak/keycloak.dockerfile:3]: Is there a source for this plugin?

File: packages/react-components/lib/reports/user-report/user-login-failure-report-table.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  created: string; //date
+  ip_address: string;
+  payload: string | unknown;
+  client_id: string;
+  username: string;
+  error: string;
+}[];
+
+export interface UserLoginFailureReportTable extends DefaultLogTableProps {
+  rows: UserLoginFailureRowsType | [];
```
Comment (koonpeng) [inline packages/react-components/lib/reports/user-report/user-login-failure-report-table.tsx:18]: Why do we need `[]`? I thought `UserLoginFailureRowsType` already allows an empty array.

File: packages/reporting-server/models/auth_events.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from tortoise import fields, models
+from tortoise.contrib.pydantic import pydantic_model_creator
+
+
+class AuthEvents(models.Model):
+    id = fields.IntField(pk=True)
+    # user = fields.ForeignKey(User, null=True)
+    username = fields.TextField(null=True)
```
Comment (koonpeng) [inline packages/reporting-server/models/auth_events.py:8]: Should we use `TextField` or `CharField`? I'm guessing `CharField` is more performant.

File: packages/reporting-server/rest_server/routers/log.py [hunk 4]
```diff
@@ -32,3 +33,20 @@ async def write_rmf_server_logs(body: list):
     except Exception as e:
         print(e)
         raise HTTPException(503, "cannot create the rmfserver log" + str(e)) from e
+
+
+# Will receive information from keacloak only
```
Comment (koonpeng) [inline packages/reporting-server/rest_server/routers/log.py:38]: typo: keacloak -> keycloak

**Review by koonpeng — APPROVED** [review]

File: example-deployment/docker/keycloak/keycloak.dockerfile (hunk 1 above)
Comment (matiasbavera) [reply example-deployment/docker/keycloak/keycloak.dockerfile:3]: Added the source as a comment

File: packages/reporting-server/models/auth_events.py (hunk 3 above)
Comment (matiasbavera) [reply packages/reporting-server/models/auth_events.py:8]: Yeah, you're right. Updating

File: packages/reporting-server/rest_server/routers/log.py (hunk 4 above)
Comment (matiasbavera) [reply packages/reporting-server/rest_server/routers/log.py:38]: Thank you

File: packages/react-components/lib/reports/user-report/user-login-failure-report-table.tsx (hunk 2 above)
Comment (matiasbavera) [reply packages/react-components/lib/reports/user-report/user-login-failure-report-table.tsx:18]: Yeah, you're right, we don't. Thanks for pointing this!

**Pull request merged**